### PR TITLE
Refactor to change ACI CNI inventory fields, tasks and SGs

### PR DESCRIPTION
**Pull request by apoorva11029**
1. Remove the Ansible task to set MTU for node network
2. Delete the accProvisiontar unpacked for preprocessing
3. Open all TCP and UDP master and worker ports
4. Remove FIP association for master and worker nodes

### CODE REVIEW

File: upi/openstack/update_ign.py [hunk 1]
```diff
@@ -10,18 +11,22 @@
 #According to the number of the master count, create the JSON files, and add hostname/network-scripts.
 #According to the number of the worker count, create the JSON files, and add hostname/network-scripts.
 
-# Read inventory.yaml for CiscoACI CNI variables
-with open("inventory.yaml", 'r') as stream:
+# Read inventory.yaml for CiscoACI CNI variable
+original_inventory = "inventory.yaml"
+processed_inventory = "inventory.yaml"
```
Comment (abhis2112) [inline upi/openstack/update_ign.py:16]: Just a trivial comment, if the name is the same, and nothing else here do we need an extra variable here? because there is no difference here

Comment (apoorva11029) [reply upi/openstack/update_ign.py]: Modified accordingly, keeping the two variables though in case we don't want to change the user-created inventory

File: upi/openstack/021_network.yaml [hunk 2]
```diff
@@ -6,35 +6,29 @@
...
+      cidr: "{{ aci_cni['network_interfaces']['opflex']['subnet'] }}"
+      allocation_pool_start: "{{ aci_cni['network_interfaces']['opflex']['subnet'] | next_nth_usable(10) }}"
+      allocation_pool_end: "{{ aci_cni['network_interfaces']['opflex']['subnet'] | ipaddr('last_usable') }}"
 
   - name: 'Set the ACI containers cluster subnet tag'
     command:
       cmd: "openstack subnet set --tag {{ cluster_id_tag }} {{ os_aci_containers_subnet }}"
-    when: os_networking_type == "CiscoACI"
 
   - name: 'Set MTU for the ACI containers node  network'
```
Comment (snaiksat) [inline upi/openstack/021_network.yaml:28]: Though not part of this PR, this name seems incorrect, it should be "Set MTU for the ACI containers network", no?

File: upi/openstack/inventory.yaml [hunk 3]
```diff
@@ -34,11 +33,16 @@ all:
...
+        acc_provision_tar: /home/noiro/openupi/aci_deployment.yaml.tar.gz
+        network_interfaces:
+          node:
+            name: ens3
+            vrf: uni/tn-common/ctx-sauto_l3out-1_vrf
+            # no default MTU setting for node network
+          opflex: 
+            name: ens4
+            subnet: '16.11.0.0/24'
+            mtu: 1600
```
Comment (snaiksat) [inline upi/openstack/inventory.yaml:46]: I believe this should be 1500, since the neutron network is being set to 1500 MTU by default.

File: upi/openstack/021_network.yaml (hunk 2 above)
Comment (apoorva11029) [reply upi/openstack/021_network.yaml]: Correct, changed!

File: upi/openstack/inventory.yaml (hunk 3 above)
Comment (apoorva11029) [reply upi/openstack/inventory.yaml]: I had set it to another value to test the functionality as 1500 is the default anyway, but yes set it back to 1500 to show a default example.

File: upi/openstack/02_network.yaml [hunk 4]
```diff
@@ -156,7 +168,8 @@
     command:
       cmd: "openstack floating ip set --port {{ os_port_ingress }} {{ os_ingress_fip }}"
 
-  - name: 'Set MTU for the first neutron network'
+
+  - name: 'Create SNAT subnet'
     command:
-      cmd: "openstack networrk set {{ os_network }} --mtu {{ neutron_network_mtu }}"
+      cmd: "neutron subnet-create {{ os_external_network }} {{ aci_cni['snat_host_pool'][subnet] }} --name snat-subnet-pool --disable-dhcp --gateway {{ aci_cni['snat_host_pool']['gateway'] }} --apic:snat_host_pool True"
```
Comment (abhis2112) [inline upi/openstack/02_network.yaml:174]: as per hierarchy should be aci_cni['network_interfaces']['node']['snat_host_pool']['subnet'] and aci_cni['network_interfaces']['node']['snat_host_pool']['gateway'] right?

Comment (apoorva11029) [reply upi/openstack/02_network.yaml]: Good catch, fixed!

File: upi/openstack/01_security-groups.yaml [hunk 5]
```diff
@@ -31,6 +31,34 @@
       security_group: "{{ os_sg_master }}"
       protocol: icmp
 
+  - name: 'Create master-sg rule for all TCP ports'
+    os_security_group_rule:
+      security_group: "{{ os_sg_master }}"
+      protocol: tcp
+      remote_ip_prefix: 0.0.0.0/0
```
Comment (snaiksat) [inline upi/openstack/01_security-groups.yaml:38]: remote_ip_prefix: "{{ cluster_network_cidrs }}"

File: upi/openstack/01_security-groups.yaml [hunk 6]
```diff
@@ -31,6 +31,34 @@
...
+    when: os_networking_type == "CiscoACI"
+
+  - name: 'Create master-sg rule for all UDP ports'
+    os_security_group_rule:
+      security_group: "{{ os_sg_master }}"
+      protocol: udp
+      remote_ip_prefix: 0.0.0.0/0
```
Comment (snaiksat) [inline upi/openstack/01_security-groups.yaml:45]: ditto

File: upi/openstack/01_security-groups.yaml [hunk 7]
```diff
@@ -31,6 +31,34 @@
...
+    when: os_networking_type == "CiscoACI"
+
+  - name: 'Create worker-sg rule for all TCP ports'
+    os_security_group_rule:
+      security_group: "{{ os_sg_worker }}"
+      protocol: tcp
+      remote_ip_prefix: 0.0.0.0/0
```
Comment (snaiksat) [inline upi/openstack/01_security-groups.yaml:52]: ditto

File: upi/openstack/01_security-groups.yaml [hunk 8]
```diff
@@ -31,6 +31,34 @@
...
+    when: os_networking_type == "CiscoACI"
+
+  - name: 'Create worker-sg rule for all UDP ports'
+    os_security_group_rule:
+      security_group: "{{ os_sg_worker }}"
+      protocol: udp
+      remote_ip_prefix: 0.0.0.0/0
```
Comment (snaiksat) [inline upi/openstack/01_security-groups.yaml:59]: ditto

File: upi/openstack/02_network.yaml [hunk 9]
```diff
@@ -156,7 +168,8 @@
     command:
       cmd: "openstack floating ip set --port {{ os_port_ingress }} {{ os_ingress_fip }}"
 
-  - name: 'Set MTU for the first neutron network'
+
+  - name: 'Create SNAT subnet'
```
Comment (snaiksat) [inline upi/openstack/02_network.yaml:172]: Apologies that i suggested adding this, but since this is not per cluster, so we need to do this only once when the external network is created, and not part of this playbook. We have to do it once when we create the external network. Let's remove this.

File: upi/openstack/inventory.yaml [hunk 10]
```diff
@@ -34,11 +33,19 @@ all:
...
-      aci_containers_l3out_vrf: uni/tn-common/ctx-sauto_l3out-1_vrf
-      aci_containers_subnet_pool: subpool-openshiftupi
-      aci_containers_prefix_length: 27
+      aci_cni:
+        acc_provision_tar: /home/noiro/openupi/aci_deployment.yaml.tar.gz
+        network_interfaces:
+          node:
+            name: ens3
+            vrf: uni/tn-common/ctx-sauto_l3out-1_vrf
+            snat_host_pool:
```
Comment (snaiksat) [inline upi/openstack/inventory.yaml:42]: per comment earlier, lets remove this

**Comment by snaiksat** [comment]
Let's update the commit message also to remove the subnet pool thingy.

File: upi/openstack/02_network.yaml (hunk 9 above)
Comment (apoorva11029) [reply upi/openstack/02_network.yaml]: Done!

File: upi/openstack/inventory.yaml (hunk 10 above)
Comment (apoorva11029) [reply upi/openstack/inventory.yaml]: Done, thanks!

File: upi/openstack/01_security-groups.yaml (hunk 5 above)
Comment (apoorva11029) [reply upi/openstack/01_security-groups.yaml]: Changed!
